### DISCUSSION TITLE
Can we use async with @computed on model ? Adonis 5

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#1599] [Help] [answered]
Hi,
```javascript
@computed()
  public get callResponseRate () {
   return some value from helper `commonHelper.expertResponceRate(this.id)`
  }
```
My helper code is 
```
public async expertResponceRate(expertId){
    return return some database calculated value.
  }
```
@thetutlage I am not getting value of callResponseRate column. Is there any easy way for this.

**Comment by thetutlage** [comment, marked as answer]
Nope, computed properties cannot be async and it is a design constraint since computed properties are evaluated during model serialization and you cannot have async serialization.

I suggest using hooks for this

**Reply by pk4all** [reply]
@thetutlage Thanks.
